lib/src/execute.rs

```diff
@@ -342,3 +344,40 @@ fn configure_wasmtime() -> wasmtime::Config {
 
     config
 }
+
+struct CpuTimer<F>
+where
+    F: Future<Output = Result<(), Trap>>,
+{
+    future: Pin<Box<F>>,
+    duration: Duration,
+}
+
+impl<F> CpuTimer<F>
+where
+    F: Future<Output = Result<(), Trap>>,
+{
+    pub fn new(future: Pin<Box<F>>) -> Self {
+        Self {
+            future,
+            duration: Duration::default(),
+        }
+    }
+}
+
+impl<F> Future for CpuTimer<F>
+where
+    F: Future<Output = Result<(), Trap>>,
+{
+    type Output = (Result<(), Trap>, Duration);
+    fn poll(mut self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<Self::Output> {
+        let now = Instant::now();
+        let output = self.future.as_mut().poll(ctx);
+        let poll_time = Instant::now().duration_since(now);
+        self.duration += poll_time;
```

Comment:
```suggestion
        self.duration += now.elapsed();
```
take it or leave it, but i'm like 95% sure this is a correct suggestion...